libs/leon-sans-react/src/domain/WreathSans.ts

```diff
@@ -0,0 +1,320 @@
+import gsap, { Power0, Power3 } from 'gsap';
+import { CHARSET, ModelData } from 'leonsans';
+import LeonSans from 'leonsans/src/leonsans';
+import * as PIXI from 'pixi.js';
+
+import { randomIdx } from '../utils';
+
+const TYPO_EASING = Power0.easeNone;
+const TYPO_DRAWING_DURATION = 1;
+
+const LEAVES_EASING = Power3.easeOut;
+const LEAVES_DRAWING_SPEED = 1;
+const LEAVES_DRAWING_DELAY = TYPO_DRAWING_DURATION - 0.05;
+
+const ORNAMENT_SOURCE_NAMES = [
+  'ball_1.svg',
+  'ball_2.svg',
+  'candy.svg',
+  'fruit_1.svg',
+  'fruit_2.svg',
+  'pinecone_1.svg',
+  'pinecone_2.svg',
+  'poinsettia_1.svg',
+  'poinsettia_2.svg',
+  'ribbon.svg',
+  'star.svg',
+];
+const ORNAMENT_PROBABILITY = 0.15;
+const ORNAMENT_PROBABILITY_INCREASE = 0.24;
+const ORNAMENT_SCALE = 0.28;
+const ORNAMENT_STAR_SCALE = 0.5;
+const ORNAMENT_RADIUS = 50;
+
+type WreathSansProps = {
+  canvas: HTMLCanvasElement;
+  renderer: PIXI.Renderer;
+  stage: PIXI.Container;
+  graphics: PIXI.Graphics;
+  leon: LeonSans;
+  pixelRatio: number;
+};
+
+export default class WreathSans {
+  canvas: HTMLCanvasElement;
+  renderer: PIXI.Renderer;
+  stage: PIXI.Container;
+  graphics: PIXI.Graphics;
+  leon: LeonSans;
+  pixelRatio: number;
+  leafSources: PIXI.SpriteSource[];
+  ornamentSources: PIXI.SpriteSource[];
+  containers: PIXI.Container[];
+
+  constructor(props: WreathSansProps) {
+    this.canvas = props.canvas;
+    this.renderer = props.renderer;
+    this.stage = props.stage;
+    this.graphics = props.graphics;
+    this.leon = props.leon;
+    this.pixelRatio = props.pixelRatio;
+    this.containers = [];
+    this.leafSources = [];
+    this.ornamentSources = [];
+
+    this.loadAssets().then(() => {
+      this.redraw();
+    });
+  }
+
+  insertText(text: string, idx: number) {
+    // check text
+    if (text.length !== 1 || !(CHARSET.includes(text) || ' \n'.includes(text))) return;
+
+    // add text
+    this.leon.text =
+      this.leon.text.slice(0, idx) + text + this.leon.text.slice(idx);
+
+    // recalculate position of new text
+    const x = (this.canvas.clientWidth - this.leon.rect.w) / 2;
+    const y = (this.canvas.clientHeight - this.leon.rect.h) / 2;
+    this.leon.position(x, y);
+
+    // update paths
+    this.leon.updateDrawingPaths();
+
+    // draw
+    // FIXME : \n 처리
+    if (text === '\n') {
+      this.updatePositions();
+      return;
+    }
+    const lineBreak = this.leon.text.slice(0, idx).split('\n').length - 1;
+    this.drawTypo(this.leon.data[idx - lineBreak]);
+    this.drawLeaves(
+      this.leon.data[idx - lineBreak],
+      this.makeContainer(idx - lineBreak),
+    );
+    this.updatePositions();
```

Comment:
```suggestion
    // 새로 써진 글자만 다시 애니메이팅하기
    this.drawTypo(this.leon.data[idx - lineBreak]);
    this.drawLeaves(
      this.leon.data[idx - lineBreak],
      this.makeContainer(idx - lineBreak),
    );
    this.updatePositions();
```

이해를 위한 주석 추가. 뭔가 draw~~~ 라는 이름은 항상 애니메이션을 포함한다는 식으로 이해하겠습니다.